src/confd-entry.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+if [[ "$BALENA_USE_CONFD" = "1" ]]; then
+  # When this environment variable is set, we need to pull configuration from etcd.
+  # The image is expected to have an environment file template located in /etc/confd,
+  # and this template should be configured to rencer the actual file to /balena/env.
+  /usr/local/bin/confd -onetime -node http://172.17.42.1:4001 || exit 1
```

Comment:
We neet to take into account the env backend as well here. Something along the lines of https://github.com/balena-io/open-balena-base/blob/5a5dc6b9ef510966b86068e105de6c9115c22bad/src/confd.service#L12-L15

```
'if [[ $CONFD_BACKEND = "ENV" ]]; \
then /usr/local/bin/confd -onetime -confdir=/usr/src/app/config/confd_env_backend -backend env; \
else /usr/local/bin/confd -onetime -confdir=/etc/confd -node=http://172.17.42.1:4001; \
fi'
```

Reply:
For what kind of environments?
If we can define the configuration with env variables, it's better to set them directly on the container - it's the idea.

Reply:
With these changes, we can also remove 2 configuration templates from our project repos.

Reply:
To rephrase...
For now, we consider a model that we have one main process. And it inherits all env variables set on the container.

Reply:
>In the kubernetes world, containers bare injected envvironment variables with KUBERNETES_ or SERVICE_NAME prefixes.

Not sure what you mean exactly... Environment variables have names exactly how they are defined in a deployment spec, and there are no restrictions on how to name them...

Reply:
@roman-mazur that's not entirely true. There are some additional (injected) envvars with the patterns I mentioned above. An example: when a Pod is run on a Node, the kubelet adds a set of environment variables for each active Service.
Some additional context:
https://github.com/kubernetes/kubernetes/blob/2bd9643cee5b3b3a5ecbd3af49d09018f0773c77/pkg/kubelet/envvars/envvars.go#L32-L62

Reply:
Ok, got it. I meant the variables with values produced by k8s itself.

Reply:
As discussed in the arch call, we ll be removing confd ENV backend interface. Resolving